src/analysis/transcendental/liouville.lean

```diff
@@ -0,0 +1,529 @@
+  { cases le_or_gt 1 B1,
+    { use 1, split, norm_num, split, linarith, split, norm_num, linarith },
+    { use B1, exact ⟨hB11, ⟨hB12, ⟨le_of_lt h, hB13⟩⟩⟩ }},
+  refine ⟨B, hB1, hB2, hB3, λ (x : ℝ) (hx : abs (α - x) < B), h_B0_root x _ ⟩,
+  linarith
+end
+
+lemma exists_deriv_eq_slope_of_polynomial_root (α : ℝ) (f : polynomial ℝ) (h_α_root : f.eval α = 0)
+  (x : ℝ) (h : f.eval x ≠ 0) :
+  ∃ x₀, α - x = - ((f.eval x) / (f.derivative.eval x₀))
+    ∧ f.derivative.eval x₀ ≠ 0
+    ∧ abs (α - x₀) < abs (α - x)
+    ∧ abs (x - x₀) < abs (α - x) :=
+begin
+  have h₀ : x ≠ α, { intro h₁, rw ← h₁ at h_α_root, rw h_α_root at h, tauto },
+  rcases ne_iff_lt_or_gt.1 h₀ with h_α_gt | h_α_lt,
+  { -- When `x < α`
+    have h_cont : continuous_on (λ x, f.eval x) (Icc x α) := f.continuous_eval.continuous_on,
+    have h_diff : differentiable_on ℝ (λ x, f.eval x) (Ioo x α) :=
+      differentiable.differentiable_on f.differentiable,
+    rcases (exists_deriv_eq_slope (λ x, f.eval x) h_α_gt h_cont h_diff) with ⟨x₀, x₀_range, hx₀⟩,
+    rw polynomial.deriv at hx₀,
+    change eval x₀ f.derivative = (eval α f - eval x f) / (α - x) at hx₀,
+    rw [h_α_root, zero_sub] at hx₀,
+    replace hx₀ := hx₀.symm,
+    have h_Df_nonzero : f.derivative.eval x₀ ≠ 0 := hx₀.symm ▸ λ hc, h
+      begin
+      rwa [hc, neg_div, neg_eq_zero, div_eq_iff (show α - x ≠ 0, by linarith), zero_mul] at hx₀ end,
+    use x₀,
+    split,
+    { symmetry, rw ← neg_div, rw div_eq_iff at hx₀ ⊢, rwa mul_comm,
+      exact h_Df_nonzero,
+      rw sub_ne_zero, exact h₀.symm },
+    apply and.intro h_Df_nonzero,
+    rw mem_Ioo at x₀_range,
+    rw [abs_of_pos (sub_pos.mpr h_α_gt), abs_of_pos (sub_pos.mpr x₀_range.2),
+      abs_of_neg (sub_lt_zero.mpr x₀_range.1)],
+    split; linarith },
+  { -- When `α < x`
+    have h_cont : continuous_on (λ x, f.eval x) (Icc α x) := f.continuous_eval.continuous_on,
+    have h_diff : differentiable_on ℝ (λ x, f.eval x) (Ioo α x):=
+      differentiable.differentiable_on f.differentiable,
+    rcases (exists_deriv_eq_slope (λ x, f.eval x) h_α_lt h_cont h_diff) with ⟨x₀, x₀_range, hx₀⟩,
+    rw polynomial.deriv at hx₀,
+    change eval x₀ f.derivative = (eval x f - eval α f) / (x - α) at hx₀,
+    rw [h_α_root, sub_zero] at hx₀,
+    replace hx₀ := hx₀.symm,
+    have h_Df_nonzero : f.derivative.eval x₀ ≠ 0 := hx₀.symm ▸ λ hc, h
+      begin rwa [hc, div_eq_iff (show x - α ≠ 0, by linarith), zero_mul] at hx₀ end,
+    use x₀,
+    split,
+    { symmetry, rw ← neg_div, rw div_eq_iff at hx₀ ⊢,
+      {rw hx₀, ring },
+      { exact h_Df_nonzero },
+      { rwa sub_ne_zero }},
+    apply and.intro h_Df_nonzero,
+    rw mem_Ioo at x₀_range,
+    rw [abs_of_neg (sub_lt_zero.mpr x₀_range.1), abs_of_neg (sub_lt_zero.mpr h_α_lt),
+      abs_of_pos (sub_pos.mpr x₀_range.2)],
+    split; linarith }
+end
+
+lemma exists_pos_real_of_irrational_root (α : ℝ) (hα : irrational α) (f : polynomial ℤ)
```

Comment:
This lemma seems quite specialized, so it makes sense to keep it in this file. (Just a remark, not a question. Only pointing out that the sequence of lemmas that should be moved to another file has now stopped.)